crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -618,6 +616,17 @@ pub fn as_parent_path_with(
         .collect()
 }
 
+pub fn is_in_try(ast_path: &AstNodePath<AstParentNodeRef<'_>>) -> bool {
+    ast_path
+        .iter()
+        .fold(false, |in_try, ast_ref| match ast_ref.kind() {
+            AstParentKind::ArrowExpr(ArrowExprField::Body) => false,
+            AstParentKind::Function(FunctionField::Body) => false,
+            AstParentKind::TryStmt(TryStmtField::Block) => true,
+            _ => in_try,
+        })
```

Comment:
`rev().find_map()` might also work well and might be a bit faster

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map